src/App.tsx

```diff
@@ -1,10 +1,39 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useContext, useEffect, useRef } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+
+import { USER_ID, getTodos } from './api/todos';
+import { ActionType } from './types/ReducerTypes';
+
+import { DispatchContext, StateContext } from './store/reducer';
+
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
 
 export const App: React.FC = () => {
+  const { todos } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+
+    getTodos()
+      .then(result => dispatch({ type: ActionType.SetTodos, payload: result }))
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+
+        setTimeout(() => {
```

Comment:
[Not critical]
It also makes sense to clear this timeout when effect is cleared:
```
useEffect(() => {
  // set timeout
  return () => {
    // clear timeour
  };
}, []);
```